src/file/file.service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@nestjs/common';
+import { MAX_FILE_SIZE } from './constants/file.constants';
+import { Stream } from 'stream';
+import { createWriteStream } from 'fs';
+import { join } from 'path';
+
+@Injectable()
+export class FileService {
+  async uploadFile(file: Express.Multer.File) {
+    if (file.size <= MAX_FILE_SIZE) {
+      return { message: 'File uploaded successfully', file };
+    } else {
```

Comment:
Logical error

if the file is under MAX_SIZE - function will newer work
If file size is bigger then allowed - we do logic

You need to do the smallest if as you can
For example wright implementation here will be:

```
if (file.size >= MAX_FILE_SIZE) {
  throw new Error("File size is more then allowed");
}
```
and the rest code will not be in "else" case

Reply:
You can also controll this size validation using configuration for the FileInterceptor

Reply:
done